src/android.ts

```diff
@@ -46,10 +46,14 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
         objectionProcesses: [],
 
         awaitAdb: async () => {
-            const adbIsStarted = await retryCondition(
-                async () => (await execa('adb', ['get-state'], { reject: false })).exitCode === 0,
-                100
-            );
+            const adbIsStarted = await retryCondition(async () => {
+                const { exitCode, stdout } = await execa(
+                    'adb',
+                    ['wait-for-device', 'shell', 'getprop', 'dev.bootcomplete'],
```

Comment:
Just to document this, because it wasn't apparent to me:

If we `wait-for-device`, we don't need the `get-state` check anymore. Here is the output from `adb help`:

```
 wait-for[-TRANSPORT]-STATE...
     wait for device to be in a given state
     STATE: device, recovery, rescue, sideload, bootloader, or disconnect
     TRANSPORT: usb, local, or any [default=any]
 get-state                print offline | bootloader | device
```

We really should have made sure that `get-state` prints `device` (and not just wait for the non-zero exit code to begin with. `wait-for-device` does just that, but instead of us calling `get-state` again and again, it does that internally.